### PR TITLE
Move linux guest test to regression tests

### LINKED ISSUE #1910: lib/guest unit tests require an ESX to run
Unit tests should not reach to external resources like an ESX. Currently the tests under `lib/guest` all require an ESX URL to be set:

```
HMAHMOOD-MBPRO:guest hmahmood$ go test . -v
=== RUN   TestNewLinuxGuest
--- SKIP: TestNewLinuxGuest (0.00s)
	env.go:27: Skipping: No test ESX URL defined
PASS
ok  	github.com/vmware/vic/lib/guest	0.014s
```
The tests should be rewritten to mock out the external resources, or we should move them to integration tests, or we could use vcsim for them.

### PR DESCRIPTION
**Pull request by sgairo**
Fixes #1910 

Removes linux_test.go from lib/guest and adds a regression test.

### CODE REVIEW

**Comment by sgairo** [comment]
@mhagen-vmware  could you check this out please :)

File: tests/test-cases/Group7-Regression/7-1-Regression.robot [hunk 1]
```diff
@@ -7,4 +7,6 @@ Default Tags  regression
 
 *** Test Cases ***
 Regression test
-    Run Regression Tests
\ No newline at end of file
+    Run Regression Tests
+    	${out}= Run govc vm.info ${vch-name}
+	Should contain ${out} Other 3.x or later Linux (64-bit)
```
Comment (mhagen-vmware) [inline tests/test-cases/Group7-Regression/7-1-Regression.robot]: Can you line these up with the existing keyword?

Comment (mhagen-vmware) [inline tests/test-cases/Group7-Regression/7-1-Regression.robot]: Spacing is off as well.  Need double spaces between keyword and parms. At least after your Run command and Should Contain

Comment (sgairo) [inline tests/test-cases/Group7-Regression/7-1-Regression.robot]: Thanks!

**Comment by sgairo** [comment]
@hmahmood Could I get your input please?

File: tests/test-cases/Group7-Regression/7-1-Regression.robot [hunk 2]
```diff
@@ -7,4 +7,6 @@ Default Tags  regression
 
 *** Test Cases ***
 Regression test
-    Run Regression Tests
\ No newline at end of file
+    Run Regression Tests
+    ${out}=  Run  govc vm.info ${vch-name}
+    Should contain  ${out}  Other 3.x or later Linux (64-bit)
```
Comment (anchal-agrawal) [inline tests/test-cases/Group7-Regression/7-1-Regression.robot]: `Contain` so we're consistent.

Comment (mhagen-vmware) [inline tests/test-cases/Group7-Regression/7-1-Regression.robot]: lgtm

after Anchal's cap comment :)

Comment (sgairo) [inline tests/test-cases/Group7-Regression/7-1-Regression.robot]: alright, done. Thanks @anchal-agrawal !

**Comment by anchal-agrawal** [comment]
Thanks Sophie! LGTM

**Comment by sgairo** [comment]
ping

**Comment by mhagen-vmware** [comment]
lgtm

**Pull request merged**